### PR TITLE
Add AWS Foundation Security for KMS

### PR DESCRIPTION
**Pull request by spangenberg**

### CODE REVIEW

File: queries/kms/inline_policy_blocked_kms_actions.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+select arn,
+       account_id
+from aws_iam_users
+         join iam_user_violations on iam_user_violations.cq_id = aws_iam_users.cq_id
+union
+select arn,
+       account_id
+from aws_iam_roles
+         join iam_role_violations on iam_role_violations.cq_id = aws_iam_roles.cq_id
+where arn not like '%service-role/%'
```
Comment (bbernays) [inline queries/kms/inline_policy_blocked_kms_actions.sql:70]: nit: can we make this `arn:aws:iam::%:role/aws-service-role/%` instead as the existing pattern can match with something like:

`arn:aws:iam::0123456789:role/test/pathat/user-aws-service-role/session`

File: queries/kms/inline_policy_blocked_kms_actions.sql [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+               jsonb_array_elements(policy_document -> 'Statement') as statement
+          where statement ->> 'Effect' = 'Allow'
+            and statement -> 'Resource' ?| array['*',
+                      'arn:aws:kms:*:' || account_id || ':key/*',
+                      'arn:aws:kms:*:' || account_id || ':alias/*']
+            and statement -> 'Action' ?| array['*',
+              'kms:*',
+              'kms:decrypt',
+              'kms:deencrypt*',
+              'kms:reencryptfrom'] )
```
Comment (bbernays) [inline queries/kms/inline_policy_blocked_kms_actions.sql:60]: Is there some way we could simplify or remove this nearly duplicated query? Maybe make a view?

**Review by bbernays — CHANGES_REQUESTED** [review]
Overall looks good. just a couple of small things

File: foundational_security/policy.hcl [hunk 3]
```diff
@@ -429,17 +429,17 @@ policy "foundational_security" {
         description = "KMS controls"
         query "1" {
             description = "IAM customer managed policies should not allow decryption and re-encryption actions on all KMS keys"
-            query = "select 1;"
+            query = file("queries/kms/customer_policy_blocked_kms-actions.sql")
```
Comment (disq) [inline foundational_security/policy.hcl:432]: Nit:
```suggestion
            query = file("queries/kms/customer_policy_blocked_kms_actions.sql")
```
Rename the file as well

File: queries/kms/rotation_enabled_for_customer_key.sql [hunk 4]
```diff
@@ -1 +1,6 @@
-      SELECT account_id, region, arn FROM aws_kms_keys WHERE rotation_enabled = FALSE AND manager = 'CUSTOMER'
\ No newline at end of file
+select account_id,
+       region,
+       arn
+from aws_kms_keys
+where rotation_enabled is false
```
Comment (disq) [inline queries/kms/rotation_enabled_for_customer_key.sql:5]: `is false` or `is distinct from true`? (ie. if it's null, then what?)

Comment (spangenberg) [reply queries/kms/rotation_enabled_for_customer_key.sql:5]: Actually just reformatted the file, not quite sure.

File: queries/kms/inline_policy_blocked_kms_actions.sql (hunk 1 above)
Comment (spangenberg) [reply queries/kms/inline_policy_blocked_kms_actions.sql:70]: Good catch!

File: foundational_security/policy.hcl (hunk 3 above)
Comment (disq) [reply foundational_security/policy.hcl:432]: marked as resolved but I see it's still typoed

Comment (spangenberg) [reply foundational_security/policy.hcl]: Sorry, didn't push yet just did it locally.

File: queries/kms/inline_policy_blocked_kms_actions.sql (hunk 2 above)
Comment (bbernays) [reply queries/kms/inline_policy_blocked_kms_actions.sql:60]: At this time there is no way to simplify this query without taking on more complexity in a single use function. Thanks for looking into the issue.

**Review by bbernays — APPROVED** [review]
LGTM

**Pull request merged**